src/anomalib/models/efficientad/torch_model.py

```diff
@@ -308,6 +308,8 @@ def forward(self, batch: Tensor, batch_imagenet: Tensor = None) -> Tensor | dict
                 (ae_output - student_output[:, self.teacher_out_channels :]) ** 2, dim=1, keepdim=True
             )
 
+            map_st = F.pad(map_st, (4, 4, 4, 4))
```

Comment:
Without this padding the segmentation metric will decrease very much because the segmentation pixels are not aligned anymore after the next line (`map_st = F.interpolate(map_st, size=(self.input_size[0], self.input_size[1]), mode="bilinear")`).

It is similar to applying 11x11 edge filter to 100x100 image. The result will be 90x90 image with detected edges. Resizing it to 100x100 would distort the detected edges. Instead you have to pad the 90x90 image with 5 pixels on each size.

In the case of EfficientAD, the patch size is 33, so 16 pixels padding on each border. The teacher has two layers with stride 2, so we need to divide the 16 pixels padding by 2 and again by 2, which gives us 4. So 4 pixels padding on each border for the output feature maps.